InformationScripting/src/dataformat/Tuple.cpp

```diff
@@ -28,16 +28,17 @@
 
 namespace InformationScripting {
 
-Tuple::Tuple(std::initializer_list<NamedProperty> initialValues)
-	: values_{initialValues}
+Tuple::Tuple(std::initializer_list<NamedProperty> initialValues, QString tag)
+	: values_{initialValues}, tag_{tag}
 {}
 
-Tuple::Tuple(QList<NamedProperty> initialValues)
-	: values_{initialValues}
+Tuple::Tuple(QList<NamedProperty> initialValues, QString tag)
+	: values_{initialValues}, tag_{tag}
 {}
 
 QString Tuple::tag() const
 {
+	if (!tag_.isNull()) return tag_;
```

Comment:
Well I left the old behavior mostly that it is not a too invasive change. For example for ast nodes it makes sense, but we can also change it to ast.node or something. There are no drawbacks of removing the old behavior. I think we can change it, if you'd like so.